pkg/scannerv4/mappers/mappers.go

```diff
@@ -1094,3 +1275,28 @@ func vulnsEqual(a, b *claircore.Vulnerability) bool {
 		a.NormalizedSeverity == b.NormalizedSeverity &&
 		a.FixedInVersion == b.FixedInVersion
 }
+
+// dedupeAdvisories deduplicates repeat advisories out of vulnIDs and returns the result.
+// This function will only filter if ROX_SCANNER_V4_RED_HAT_CSAF is enabled; otherwise,
+// it'll just return the original slice of vulnIDs.
+// This function does not guarantee order is preserved.
+func dedupeAdvisories(vulnIDs []string, protoVulns map[string]*v4.VulnerabilityReport_Vulnerability) []string {
+	filtered := make([]string, 0, len(vulnIDs))
+	// advisories tracks the unique advisories.
+	advisories := set.NewStringSet()
+	for _, vulnID := range vulnIDs {
+		vuln := protoVulns[vulnID]
+		if vuln == nil {
+			continue
+		}
+
+		name := vuln.GetName()
+		if RedHatAdvisoryPattern.MatchString(name) && !advisories.Add(name) {
+			continue
+		}
+
+		filtered = append(filtered, vulnID)
+	}
+
+	return filtered
+}
```

Comment:
Since `mappers.go` has grown quite large, I wonder if it would make sense to move some of the score calculations (CVSS, NVD CVSS, EPSS) to a separate file within the same package, such as mappers/scores.go. This would be similar to how we’ve structured enrichers with `fetcher.go` and `parser.go`.
This can be a follow-up and doesn’t need to be included in this PR.

Reply:
Good point. Yeah, let's handle this in a followup